lib/elixir/lib/kernel.ex

```diff
@@ -1592,17 +1592,18 @@ defmodule Kernel do
 
   """
   defmacro left <> right do
-    concats = extract_concatenations({:<>, [], [left, right]})
-    quote(do: <<unquote_splicing(concats)>>)
+    concats = extract_concatenations({:<>, [], [left, right]}, __CALLER__)
+    {:<<>>, [context: :concat], concats}
```

Comment:
context is a bit overloaded in this case and we already use this key for other meaning inside the metadata in Elixir. I would suggest `:origin`, which is what we use for `try` too.